plat/common/arm/generic_timer.c

```diff
@@ -225,14 +231,13 @@ static inline uint64_t generic_timer_epochoffset(void)
  */
 void generic_timer_cpu_block_until(uint64_t until_ns)
 {
-	uint64_t now_ns, until_ticks;
+	uint64_t now_ns, now_ticks, until_ticks;
 
 	UK_ASSERT(ukplat_lcpu_irqs_disabled());
 
-	/* Record current ns and until_ticks for timer */
-	now_ns = ukplat_monotonic_clock();
-	until_ticks = generic_timer_get_ticks()
-				+ ns_to_ticks(until_ns - now_ns);
+	/* Record current ns (and ticks) and calculate until_ticks for timer */
+	now_ns      = generic_timer_monotonic_ticks(&now_ticks);
+	until_ticks = now_ticks + ns_to_ticks(until_ns - now_ns);
```

Comment:
This looks good, but I think we should move the `until_ticks` computation inside the if statement.
Without this change, we will have a conflict with the ARM virtio_mmio PR: https://github.com/unikraft/unikraft/pull/208.
See this commit as a reference: https://github.com/unikraft/unikraft/pull/208/commits/de3da1672cd476136f32ae8db061da7b96b86476.

Reply:
ok, it's done.